spacelift/resource_run.go

```diff
@@ -80,11 +154,130 @@ func resourceRunCreate(ctx context.Context, d *schema.ResourceData, meta interfa
 		variables["proposed"] = graphql.NewBoolean(graphql.Boolean(proposed.(bool)))
 	}
 
-	if err := meta.(*internal.Client).Mutate(ctx, "ResourceRunCreate", &mutation, variables); err != nil {
+	client := meta.(*internal.Client)
+	if err := client.Mutate(ctx, "ResourceRunCreate", &mutation, variables); err != nil {
 		return diag.Errorf("could not trigger run for stack %s: %v", stackID, internal.FromSpaceliftError(err))
 	}
 
-	d.SetId(mutation.ID)
+	if waitRaw, ok := d.GetOk("wait"); ok {
+		wait := expandWaitConfiguration(waitRaw.([]interface{}))
+
+		if !wait.disabled {
+			stateConf := &retry.StateChangeConf{
+				ContinuousTargetOccurence: 1,
+				Delay:                     10 * time.Second, // TODO: Delay must be a multiple of Timeout
+				MinTimeout:                10 * time.Second,
+				Pending: []string{
+					"running",
+				},
+				Target: []string{
+					"finished",
+					"unconfirmed", // Let's treat unconfirmed as the target state.
+					// It's not finished, but we don't want to wait for it because it requires confirmation from someone.
+				},
+				Refresh: checkStackStatusFunc(ctx, client, stackID, mutation.ID),
+				Timeout: d.Timeout(schema.TimeoutCreate),
+			}
+
+			finalState, err := stateConf.WaitForStateContext(ctx)
+			if err != nil {
+				if timeoutErr, ok := internal.AsError[*retry.TimeoutError](err); ok {
+					tflog.Debug(ctx, "received retry.TimeoutError from WaitForStateContext", map[string]any{
+						"stackID":       stackID,
+						"runID":         mutation.ID,
+						"lastState":     timeoutErr.LastState,
+						"expectedState": timeoutErr.ExpectedState,
+					})
+					finalState = "__timeout__"
+				} else if err == context.DeadlineExceeded {
+					tflog.Debug(ctx, "received context.DeadlineExceeded from WaitForStateContext", map[string]any{
+						"stackID": stackID,
+						"runID":   mutation.ID,
+					})
+					finalState = "__timeout__"
+				} else {
+					return diag.Errorf("failed waiting for run %s on stack %s to finish. error(%T): %+v ", mutation.ID, stackID, err, err)
+				}
+			}
 
+			switch finalState.(string) {
+			case "__timeout__":
+				if !wait.continueOnTimeout {
+					return diag.Errorf("run %s on stack %s has timed out", mutation.ID, stackID)
+				} else {
+					tflog.Info(ctx, "run timed out but continue_on_discarded=true",
```

Comment:
Should this be `continue_on_timeout`, not `continue_on_discarded`?

Reply:
ou, right, fixed. 🙇‍♂️